### ISSUE TITLE
Root Explorer: weird display and some symlink to dir treated as files

### ISSUE THREAD

**Issue by RigacciOrg** [TeamAmaze/AmazeFileManager#1497]
As you can see from the image attached, the **Root Explorer** screen displays a weird string instead of the plain file name. It seems the output of the `ls -l` command.
Also some symlink to directories are treated as files, so if I tap on `/etc` or `/sdcard` I got the **Open As** menu instead of changing directory.
Running Amaze version 3.2.2 from F-Droid, on Android Oreo 8.1.0. Root `su` is provided by Magisk 17.3
![screen1](https://user-images.githubusercontent.com/863803/47960440-7acfee00-dffc-11e8-897d-011ce65618f8.png)

**Comment by VishalNehra** [comment]
````ls -l```` command is an unreliable source for listing files. I was working for a ````stat```` implementation in the past. Let's see if I can find some time to finish it.

**Comment by EmmanuelMess** [comment]
Related #843

**Comment by RigacciOrg** [comment]
You can get different results upon the `ls` command the system defaults to. If I have the 
`/system/bin/ls -> toybox` everything works ok, but if I install BusyBox (Stericson installer) it enables `/sbin/ls -> /sbin/busybox` wich causes the problem.

Infact the outout of the two commands are quite different. In the example below you can guess that the culprit is the different date format:

```
/system/bin/ls -l /sdcard                                                                                                          
lrwxrwxrwx 1 root root 21 1970-07-14 01:38 /sdcard -> /storage/self/primary
```

```
/sbin/ls -l /sdcard                                                                                                                
lrwxrwxrwx    1 0        0               21 Jul 13  1970 /sdcard -> /storage/self/primary
```

**Comment by VishalNehra** [comment]
Does this happen on every version of busybox?

**Comment by EmmanuelMess** [comment]
@TranceLove did you check this issue was fixed at 2063? Because I just realized I made a mistake.

**Comment by VishalNehra** [comment]
@EmmanuelMess I don't understand 🤔

**Comment by TranceLove** [comment]
> @TranceLove did you check this issue was fixed at 2063? Because I just realized I made a mistake.

Seem to have no problem at all, even without this changeset. Fairphone 3 running LineageOS 16.0 (9.0).

**Comment by VishalNehra** [comment]
@TranceLove Tbh there were differences. Earlier to check for symlink file /directory we parsed the ls and from the extracted symlink path, list all files to that path's parent and find the file we extracted and check if it's directory or not (from permissions). 

Now this is just for one symlink, consider if we have 10's of symlinks in single dir. That many listing calls. 
For reference, loading /system/bin takes around 10 secs with older implementation. With stat it's almost instant. No extra listing calls.

Only difference is stat follows the path with symlink, so once you press on a symlink directory, it actually goes to that path (resulting in address bar changing to linked location). With older ls the address bar still remains to original path + symlink name
